index.js

```diff
@@ -30,26 +35,37 @@ const rateLimiting = (req, res, next, logging) => {
   const now = moment()
   const addressObject = _rlAddressToRequests[address]
   if (!addressObject) {
-    addAddressToRequests(address, 1, now)
+    addAddressToRequests(address, 1, now, now)
     return false
   }
 
   let requests = addressObject.requests
   const startRequestAt = addressObject.startRequestAt
   const diffSeconds = now.diff(startRequestAt)
-  if (diffSeconds < _interval && requests > _limit) {
-    addAddressToRequests(address, requests + 1, now)
+  const limitDenominator = (totalActiveUsers > 0 ? totalActiveUsers : 1) //1 if there is no dynamic rate limiting
+  const requestLimit = _limit/limitDenominator
+  if (diffSeconds < _interval && requests > requestLimit) {
+    console.log(address + ' broke the limit of ' + requestLimit + ' connections')
```

Comment:
Remove this printout maybe

Reply:
Yes, but we will have to add it during demonstration.